src/wallet/wallet.cpp

```diff
@@ -719,17 +719,31 @@ void CWallet::SetUsedDestinationState(WalletBatch& batch, const uint256& hash, u
     }
 }
 
-bool CWallet::IsUsedDestination(const CTxDestination& dst) const
-{
-    LOCK(cs_wallet);
-    return IsMine(dst) && GetDestData(dst, "used", nullptr);
-}
-
 bool CWallet::IsUsedDestination(const uint256& hash, unsigned int n) const
 {
+    AssertLockHeld(cs_wallet);
```

Comment:
nit, also add the EXCLUSIVE_LOCKS_REQUIRED annotation.

Reply:
done

Reply:
The annotation is supposed to go in the header file.

Reply:
is it just a no-op here? I am not a clang wizard.

Reply:
`Attributes must be attached to named declarations` oooohkay fixed

Reply:
From [clang docs](https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#frequently-asked-questions):

> Attributes are part of the formal interface of a function, and should always go in the header, where they are visible to anything that includes the header. Attributes in the .cpp file are not visible outside of the immediate translation unit, which leads to false negatives and false positives.